File: LeetCode/Python/Easy/2706-Buy-Two-Chocolates/solution.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        prices.sort()

        sum_of_two_cheapest=prices[0] + prices[1]

        return money - sum_of_two_cheapest if money >= sum_of_two_cheapest else money

class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        total=0

        p=len(prices)

        limit=(10**2) + 1

        buckets=[0 for _ in range(0, limit)]

        for price in prices:
            buckets[price]+=1

        smallest=-1

        second_smallest=-1

        for price in range(0, limit):
            if buckets[price]:
                if buckets[price]==1:
                    if smallest==-1 and second_smallest==-1:
                        smallest=price
                    elif smallest!=-1 and second_smallest==-1:
                        second_smallest=price
                    else:
                        break
                else:
                    if smallest==-1 and second_smallest==-1:
                        smallest=price

                        second_smallest=price
                    elif smallest!=-1 and second_smallest==-1:
                        second_smallest=price
                    else:
                        break

        total=smallest + second_smallest

        return money - total if money >= total else money

File: LeetCode/Rust/Easy/2706-Buy-Two-Chocolates/solution.rs
impl Solution
{
    pub fn buy_choco(mut prices: Vec<i32>, money: i32) -> i32
    {
        let mut total: i32=0;

        let n: usize=prices.len();

        let limit: u32=10u32.pow(2) + 1;

        let mut buckets: Vec<u32>=vec![0; limit as usize];

        for &price in prices.iter()
        {
            buckets[price as usize]+=1;
        }

        let mut smallest: i32=-1;

        let mut second_smallest: i32=-1;

        for price in 1..limit
        {
            let index: usize=price as usize;
            
            if buckets[index] > 0
            {
                if buckets[index]==1
                {
                    if smallest==-1 && second_smallest==-1
                    {
                        smallest=price as i32;
                    }
                    else if smallest!=-1 && second_smallest==-1
                    {
                        second_smallest=price as i32;
                    }
                    else
                    {
                        break;
                    }
                }
                else
                {
                    if smallest==-1 && second_smallest==-1
                    {
                        smallest=price as i32;

                        second_smallest=price as i32;
                    }
                    else if smallest!=-1 && second_smallest==-1
                    {
                        second_smallest=price as i32;
                    }
                    else
                    {
                        break;
                    }
                }
            }
        }

        total=smallest + second_smallest;

        let result: i32=if money - total >= 0 {money - total} else{money};

        result
    }
}

impl Solution
{
    pub fn buy_choco(mut prices: Vec<i32>, money: i32) -> i32
    {
        let mut total: i32=0;

        let n: usize=prices.len();

        prices.sort();

        let smallest: i32=prices[0];

        let second_smallest: i32=prices[1];

        total=smallest + second_smallest;

        let result: i32=if money - total >= 0 {money - total} else{money};

        result
    }
}